crates/bevy_ecs/src/system/mod.rs

```diff
@@ -225,6 +225,16 @@ pub trait IntoSystem<In, Out, Marker>: Sized {
         let name = format!("Pipe({}, {})", system_a.name(), system_b.name());
         PipeSystem::new(system_a, system_b, Cow::Owned(name))
     }
+
+    fn product<B, I, O, M>(self, system: B) -> ProductSystem<Self::System, B::System>
```

Comment:
This feels like it would be more helpfully called `join` to me. By analogy, we'll later want a `.split`.